docker/Dockerfile-envoy

```diff
@@ -12,7 +12,10 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM python:3.6-buster
+FROM frolvlad/alpine-glibc:alpine-3.14_glibc-2.33
+
+# Update to the latest
+RUN apk update && apk upgrade
```

Comment:
Will this cause the build to be non-deterministic?

Reply:
Yes, but only at docker image build time.  
I think it is an acceptable practice to do "RUN apt -y update && apt -y install xyz" to get the latest packages.

Reply:
I think it's generally better practice to make the build deterministic so we can find out if one issue appears one build. If the community continuously qualify new images, we can just update our images to the new stable ones from time to time. What's the benefit we get the latest (may not be qualified?) packages ourselves?

Reply:
+1 to @qiwzhang's comment, updating to latest packages at build time is acceptable for docker images. It is better we keep them up to date to receive security patches. If issue appears, e2e test will fail, and we will know.

Reply:
I was imagining we just need to update the stable alpine version from time to time. Like https://github.com/envoyproxy/envoy/commit/ed56db4e23e2cce943252b63c2e68be6abe5215e. 

If you guys think this is OK, I am also fine with it. Since this looks low risk to me. It's just the non-repeatable build part worries me in terms of reproducing the same artifacts and signature. 

Is this equivalent to use frolvlad/alpine-glibc:latest?

Reply:
It is similar to "frolvlad/alpine-glibc:alpine-3.14_glibc-2.33" the latest.   But we need specific glibc > 2.28 (required by envoy).  they don't have "latest" tag for a specific one.

Reply:
SG. Thanks for the explain. That's why I normally don't like depend on these community build images:)